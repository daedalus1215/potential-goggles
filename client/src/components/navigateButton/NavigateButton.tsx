import React from 'react';
import { useNavigate } from 'react-router-dom';
import IconButton from '../iconButton/IconButton';

type props = {
    icon: string;
    url: string;
    style:string;
};

const NavigateButton: React.FC<props> = ({icon, url, style}) => {
    const navigate = useNavigate();
    return <IconButton
        icon={icon}
        className={style}
        onClick={() => {
            navigate(`/${url}`);
        }}
    />;
};

export default NavigateButton;