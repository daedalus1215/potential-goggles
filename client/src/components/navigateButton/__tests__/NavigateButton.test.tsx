import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom'

import NavigateButton from '../NavigateButton';

jest.mock("react-router-dom", () => ({
    ...(jest.requireActual("react-router-dom") as any),
    useNavigate: () => jest.fn(),
}));

describe('src/components/navigateButton/__test__/NavigateButton.test.tsx', () => {
    describe('NavigateButton', () => {
        it('should render', () => {
            // Arrange & Act
            const target = render(<NavigateButton icon="bi" url="bar" style="aStyle" />);

            // Assert
            expect(target.queryByTestId('icon-button')).toBeInTheDocument();
            expect(target.queryByTestId('icon-button')).toMatchSnapshot();
        });
    });
});