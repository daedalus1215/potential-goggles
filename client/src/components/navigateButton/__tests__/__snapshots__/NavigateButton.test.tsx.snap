// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/navigateButton/__test__/NavigateButton.test.tsx NavigateButton should render 1`] = `
<ButtonWrapper
  class="baseBtn IconButton aStyle"
  data-testid="icon-button"
>
  <i
    class="bi icon"
  />
</button>
`;
